concordium-consensus/tests/consensus/ConcordiumTests/Afgjort/ABBA.hs

```diff
@@ -66,7 +66,7 @@ selectFromSeq' g s =
     (Seq.index s n, Seq.deleteAt n s, g')
 
 atParty :: Party -> Traversal' (Vec.Vector a) a
-atParty = ix . fromIntegral
+atParty x = ix $ fromIntegral x
```

Comment:
I don't know why this is needed. In general these kind of changes may affect resource usage of the function since they change the inlining opportunities. This is only in tests, but still we should refrain from making these kinds of changes. They can cause unexpected problems.

Reply:
Frankly, I did not spend time investigating the cause of the issue but the change was required due to a compilation error. I have seen similar problems in 1 or 2 other places in other repos too.

Reply:
This is due to the simplified subsumption introduced in GHC 9. https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0287-simplify-subsumption.rst#costs-and-drawbacks
This does mean that eta-expansion is required in various places.

Reply:
Thanks 👍 
Good to know, bad to not understand :)